user/TASK/shoot_task/shoot_task.c

```diff
@@ -0,0 +1,249 @@
+            // rapid fire
+            shoot.mode = SHOOT_RAPID;
+        }
+    } else {
+        pwm_target = Fric_OFF;
+        shoot.mode = SHOOT_OFF;
+    }
+    
+    //Switches control function
+    if (shoot.mode == SHOOT_READY) {
+        shoot_ready_control(&shoot.trigger_motor, &shoot.hopper_motor);
+    } else if (shoot.mode == SHOOT_SINGLE) {
+        shoot_single_control(&shoot.trigger_motor, &shoot.hopper_motor);
+    } else if (shoot.mode == SHOOT_RAPID) {
+        shoot_rapid_control(&shoot.trigger_motor, &shoot.hopper_motor);
+    } else {
+        shoot_off_control(&shoot.trigger_motor, &shoot.hopper_motor);
+    }
+}
+
+
+/**
+ * @brief Update data in the shoot struct. Handles the 36:1 gear ratio in p36
+ * @param None
+ * @retval None
+ */
+static void shoot_feedback_update(void) {
+    
+    //Accounts for gear box inside P19 
+    if (shoot.trigger_motor.pos_raw - shoot.trigger_motor.last_pos_raw > HALF_ECD_RANGE)
+    {
+        //Encoder reading went from 0 to 8192, finished one circle counterclockwise
+        shoot.trigger_motor.ecd_count--;
+    }
+    else if (shoot.trigger_motor.pos_raw - shoot.trigger_motor.last_pos_raw < -HALF_ECD_RANGE)
+    {
+        //Encoder reading went from 8192 to 0, finished one circle clockwise
+        shoot.trigger_motor.ecd_count++;
+    }
+    //Gear ratio of 36:1
+    if (shoot.trigger_motor.ecd_count == HALF_TRIGGER_RATIO)
+    {
+        shoot.trigger_motor.ecd_count = -(HALF_TRIGGER_RATIO - 1);
+    }
+    else if (shoot.trigger_motor.ecd_count == -HALF_TRIGGER_RATIO)
+    {
+        shoot.trigger_motor.ecd_count = HALF_TRIGGER_RATIO - 1;
+    }
+    shoot.trigger_motor.geared_down_pos_raw = (shoot.trigger_motor.ecd_count * FULL_ECD_RANGE + shoot.trigger_motor.pos_raw) / FULL_TRIGGER_RATIO;
+    
+}
+
+/**
+ * @brief Trigger motor locks in place while flywheels ramp up to speed
+ *        Hopper motor continues to spin to provide constant feed
+ * @param Trigger motor and hopper motor structs 
+ * @retval None
+ */
+static void shoot_ready_control(Shoot_Motor_t *trigger_motor, Shoot_Motor_t *hopper_motor)
+{
+    trigger_motor->speed_set = TRIGGER_OFF;
+    hopper_motor->speed_set = HOPPER_SPEED;
+}
+
+
+/**
+ * @brief Trigger motor moves 90 degrees to allow one ball into the launcher
+ *        Hopper motor continues to spin to provide constant feed
+ * @param Trigger motor and hopper motor structs
+ * @retval None
+ */
+static void shoot_single_control(Shoot_Motor_t *trigger_motor, Shoot_Motor_t *hopper_motor)
+{
+    //Set target to be 1/4pi from now
+    if (trigger_motor->move_flag == 0 && shoot.mode == SHOOT_SINGLE)
+    {
+        trigger_motor->geared_down_pos_set = trigger_motor->geared_down_pos_raw + TRIGGER_90_DEGS;
+        trigger_motor->cmd_time = xTaskGetTickCount();
+        trigger_motor->move_flag = 1;
+    }
+
+    //Check for whether target is reached
+    if (trigger_motor->geared_down_pos_set - trigger_motor->geared_down_pos_raw > TRIGGER_REACHED_POS_RANGE)
```

Comment:
If we're getting the difference and checking if it's within a range, should this be an absolute value? Geared_down_pos_set might also need to be refactored for clarity